extra/Addons/Macros/SelectionStartEnd.lua

```diff
@@ -0,0 +1,96 @@
+﻿-- Selection Start/End with hot-keys Ctrl+[ / Ctrl+] or via Macros Menu
+Color = far.AdvControl(far.Flags.ACTL_GETCOLOR,far.Colors.COL_HELPTOPIC)
+Selection = {}
+
+Event {
+  group="EditorEvent";
+  action=function(id,Event,param)
+    local info = editor.GetInfo()
+    if Event==far.Flags.EE_READ then
+      Selection[info.EditorID] = {
+        BlockType=far.Flags.BTYPE_STREAM,
+       	StartLine=1,
+        StartPos=1,
+       	EndLine=1,
+        EndPos=0
+      }
+    end
+    if Event==far.Flags.EE_CLOSE then 
+      Selection[info.EditorID]=nil
+    end
+  end
+}
```

Comment:
This is totally redundant.

Reply:
I have moved editor.GetInfo inside if/end, so it will calls only during file opend/close